PlateSpace/Web/src/components/dialogs/review-dialog/review-dialog.js

```diff
@@ -43,14 +51,77 @@ class ReviewDialog extends Component {
     this.props.onOkClick(
       this.state.rateValue,
       this.state.reviewValue,
+      this.state.imageUrlValue,
+      this.state.clarifaiConceptsValue,
       this.props.reviewId
     );
     this.setState({
       rateValue: this.props.rateValue,
-      reviewValue: this.props.reviewValue
+      reviewValue: this.props.reviewValue,
+      imageUrlValue: this.props.imageUrlValue
     });
   }
 
+  onDrop(files) {
+
+    const file = files[0];
+    const reader = new FileReader();
+    let imageUrl;
+    //var x = this;
+    reader.onload = (data) => {
+      let fileKey = stitchClient.authedId() + "_" + Date.now().toString() + "_" + file.name;
+      let fileData = btoa(data.target.result);
+      let fileContentType = file.type;
+      var filteredConcepts;
+      
+      const s3 = stitchClient.service('aws/s3', config.S3_SERVICE_NAME);
+      const putPromise = stitchClient.executePipeline([
+        builtins.binary('base64', fileData),
+          s3.put(config.S3_BUCKET, fileKey, "public-read", fileContentType)]);
+          
+        putPromise.then(res => {
+          console.log("AWS S3 url: ", res.result[0].location);
+          imageUrl = res.result[0].location;
+          this.setState({
+            imageUrlValue: imageUrl
+          });
+          if(imageUrl) {
+            stitchClient.executePipeline([builtins.namedPipeline('processImage', { imagePublicUrl: imageUrl })])
+           .then(res => {
+              console.log('clarifai result', res.result[0]);
+              var clarifaiResult = res.result[0].bodyJSON;
+              console.log('processImage pipeline result:', clarifaiResult);
+              var concepts = clarifaiResult.outputs[0].data.concepts;
+              console.log('Clarifai concepts before filtering:', concepts);
+              
+              concepts.forEach(function(concept) {
+                if(concept.value > config.IMAGE_RECOGNITION_CONFIDENCE_THRESHOLD) {
+                  if(!filteredConcepts) {
+                    filteredConcepts = concept.name;
+                  }
+                  else {
+                    filteredConcepts += ';' + concept.name;
+                  }
+                }
+              }, this);
+              console.log('Filtered Concepts', filteredConcepts);
+              if(filteredConcepts && filteredConcepts.length > 0)
+              this.setState({
+                clarifaiConceptsValue: filteredConcepts
+              });
+            }).catch(err => {
+          console.log("An error occurred in processImage pipeline: ", err);
```

Comment:
I'll rely on Visual Studio Code auto-indenting, you let me know if you're OK with it.